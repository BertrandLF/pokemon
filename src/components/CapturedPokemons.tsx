import React from 'react';
import './styles/CapturedPokemons.css';
import Pokemon from './Pokemon';
import PokeCard from './CapturedPokemon';

interface CapturedPokemonsProps {
  pokemons: Map<number, Pokemon>;
  handleOnClick: (pokemon: Pokemon) => void;
}

class CapturedPokemons extends React.Component<CapturedPokemonsProps> {

  cells() {
    return Array.from(this.props.pokemons.values()).map((pokemon: Pokemon) =>
      <PokeCard
        key={pokemon.id}
        pokemon={pokemon}
        handleOnClick={this.props.handleOnClick}
      />
    );
  };

  render() {
    const pokemons = this.props.pokemons;
    if (pokemons.size === 0) {
      return <section className="captured-pokemons" />
    } else {
      return (
        <section className="captured-pokemons">
          {this.cells()}
        </section>
      )
    }
  };
}

export default CapturedPokemons;