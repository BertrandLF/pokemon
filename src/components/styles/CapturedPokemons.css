.captured-pokemons {
  width: 50%;
  height: 100%;
  padding: 10px;
  background-color: #09dfb1;
  border-top-right-radius: 10px;
  border-bottom-right-radius: 10px;
  justify-content: center;
}