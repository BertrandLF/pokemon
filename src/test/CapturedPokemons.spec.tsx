import React from 'react'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CapturedPokemons from '../components/CapturedPokemons';
import { pokemonMap } from './PokemonFixture';

it('displays a list of captured pokemons', () =>{

  const onClick = () => {};
  const component = shallow(
    <CapturedPokemons
      pokemons={pokemonMap}
      handleOnClick={onClick}>
    </CapturedPokemons>
  );

  expect(toJson(component)).toMatchSnapshot();

})