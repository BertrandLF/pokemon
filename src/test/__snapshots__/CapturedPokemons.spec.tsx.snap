// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a list of captured pokemons 1`] = `
<section
  className="captured-pokemons"
>
  <CapturedPokemon
    handleOnClick={[Function]}
    key="54"
    pokemon={
      Pokemon {
        "id": 54,
        "name": "psyduck",
        "sprite": "https://someUrl/someImage.png",
        "type": "water",
      }
    }
  />
  <CapturedPokemon
    handleOnClick={[Function]}
    key="1"
    pokemon={
      Pokemon {
        "id": 1,
        "name": "balbusaur",
        "sprite": "https://someUrl/someOtherImage.png",
        "type": "poison",
      }
    }
  />
</section>
`;
